Bottom: 378432a42897fbab2844f38f5f82dcbe9d0d02bc
Top:    c0bb31377e4893f58a51a2685232067828159eb8
Author: Ian Kent <ikent@redhat.com>
Date:   2013-04-04 19:05:21 +0800

Refresh of autofs-5.0.7-syncronize-handle_mounts-shutdown-2.patch

---

diff --git a/daemon/automount.c b/daemon/automount.c
index d9d6d44..6082a8c 100644
--- a/daemon/automount.c
+++ b/daemon/automount.c
@@ -1285,14 +1285,8 @@ static int do_hup_signal(struct master *master, time_t age)
 	nfs_mount_uses_string_options = check_nfs_mount_version(&vers, &check);
 
 	master_mutex_lock();
-	/* Already shutdown or no mounts ? */
-	if (list_empty(&master->mounts)) {
-		status = pthread_mutex_unlock(&mrc.mutex);
-		if (status)
-			fatal(status);
-		return;
-	}
-	if (master->reading) {
+	/* Already doing a map read or shutdown or no mounts */
+	if (master->reading || list_empty(&master->mounts)) {
 		status = pthread_mutex_unlock(&mrc.mutex);
 		if (status)
 			fatal(status);
